src/zkevm_specs/evm/execution/comparator.py

```diff
@@ -0,0 +1,51 @@
+from ...util import FQ
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+
+
+def cmp(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    is_eq, is_gt = instruction.pair_select(opcode, Opcode.EQ, Opcode.GT)
+
+    a = instruction.stack_pop()
+    b = instruction.stack_pop()
+    c = instruction.stack_push()
+
+    print("limbs", a, b, c)
+    # swap a and b if the opcode is GT
+    (aa, bb) = (b, a) if is_gt == 1 else (a, b)
+
+    # decode RLC to bytes for a and b
+    a8s = aa.le_bytes
+    b8s = bb.le_bytes
+    c8s = c.le_bytes
+
+    a_lo = instruction.bytes_to_fq(a8s[:16])
+    a_hi = instruction.bytes_to_fq(a8s[16:])
+    b_lo = instruction.bytes_to_fq(b8s[:16])
+    b_hi = instruction.bytes_to_fq(b8s[16:])
+    cc = instruction.bytes_to_fq(c8s[:31])
```

Comment:
Perhaps we don't need this because in the circuit implementation `result` only takes single cell.